src/Microsoft.Health.Fhir.Core/Features/Search/Registry/SearchParameterStatusManager.cs

```diff
@@ -22,82 +25,143 @@ public class SearchParameterStatusManager : IRequireInitializationOnFirstRequest
+                                // Registry converter information is different, re-check parameter support.
+                                isSupported = _searchParameterSupportResolver.IsSearchParameterSupported(p);
+                            }
+
+                            isSearchable = isSupported && result.Status.IsSearchable();
+
+                            // If this is a newer build, update the registry
+                            if (typeConvertersBuildDate > result.TypeConvertersBuildDate)
+                            {
+                                result.TypeConvertersHash = typeConvertersHash;
+                                result.TypeConvertersBuildDate = typeConvertersBuildDate;
+
+                                if (isSupported && result.Status == SearchParameterStatus.Disabled)
+                                {
+                                    result.Status = SearchParameterStatus.Supported;
+                                }
+
+                                updatedParameters.Add(result);
+                            }
+                        }
+
+                        if (p.IsSearchable != isSearchable ||
+                            p.IsSupported != isSupported ||
+                            p.IsPartiallySupported != result.IsPartiallySupported)
+                        {
+                            p.IsSearchable = isSearchable;
+                            p.IsSupported = isSupported;
+                            p.IsPartiallySupported = result.IsPartiallySupported;
+
+                            updated.Add(p);
+                        }
                     }
-                }
-                else
-                {
-                    newParameters.Add(new ResourceSearchParameterStatus
+                    else
                     {
-                        Uri = p.Url,
-                        LastUpdated = Clock.UtcNow,
-                        Status = SearchParameterStatus.Supported,
-                    });
+                        var isNewDatabase = parameters.Count == 0;
 
-                    p.IsSearchable = false;
+                        // Check if this parameter is now supported.
+                        if (builtinParameters.TryGetValue(p.Url, out var builtinResult))
+                        {
+                            p.IsSupported = builtinResult.Status.IsSupported();
+                        }
+                        else
+                        {
+                            // Registry converter information is different, re-check parameter support.
+                            p.IsSupported = _searchParameterSupportResolver.IsSearchParameterSupported(p);
+                        }
 
-                    // Check if this parameter is now supported.
-                    p.IsSupported = _searchParameterSupportResolver.IsSearchParameterSupported(p);
+                        p.IsSearchable = p.IsSupported && isNewDatabase;
+                        updated.Add(p);
```

Comment:
For my understanding, why isn't there a check similar to the one on lines 118-120 before adding `p` to `updated` here?

Reply:
The above check only adds it if it has changed, this one is always new